src/pydap/model.py

```diff
@@ -287,18 +375,15 @@ class StructureType(DapType, Mapping):
     """A dict-like object holding other variables."""
 
     def __init__(self, name, attributes=None, **kwargs):
-        DapType.__init__(self, name, attributes, **kwargs)
+        super(StructureType, self).__init__(name, attributes, **kwargs)
 
-        # emulate a simple ordered dict
-        self._keys = []
-        self._dict = {}
+        # allow some keys to be hidden:
+        self._visible_keys = []
```

Comment:
That would be great but there are one with ``SequenceType``. One might want to slice based on field ``'a'`` but get ``['b', 'c']``:
``
data = sequence[sequence['a'] > 0.0]['b', 'c']
for s in data.iterdata():
    print(s)  # This is when data is retrieved and sliced.
``
Because a ``SequenceType`` is expecting streaming data, this slicing cannot be done now but later. This means that ``'a'`` must carried around until actual data is passed. This is burdensome but it seems that changing this at the moment would be a major undertaking.